File: Documentation/5.4/Samples/python/ClientApi/Session/HowTo/DocumentExists.py
from examples_base import ExampleBase


class Exists:
    # region exists_1
    def exists(self, key: str) -> bool: ...

    # endregion


class DocumentExists(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_document_exists(self):
        with self.embedded_server.get_document_store("DocumentExists") as store:
            with store.open_session() as session:
                # region exists_2
                exists = session.advanced.exists("employees/1-A")

                if exists:
                    ...  # Document 'employees/1-A' exists
                # endregion

File: Documentation/5.4/Samples/php/ClientApi/Session/HowTo/DocumentExists.php
<?php

namespace RavenDB\Samples\ClientApi\Session\HowTo;

use RavenDB\Documents\DocumentStore;

interface ExistsInterface
{
    # region exists_1
    public function exists(?string $id): bool;
    # endregion
}


class DocumentExists
{
    public function example(): void
    {
        $store = new DocumentStore();
        try {
            $session = $store->openSession();
            try {
                # region exists_2
                $exists = $session->advanced()->exists("employees/1-A");

                if ($exists)
                {
                    // document 'employees/1-A exists
                }
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
